contributing.md

```diff
@@ -0,0 +1,369 @@
+# Contributing to TensorLake
+
+Thank you for your interest in contributing to TensorLake! We welcome contributions from the community and are excited to see what you'll build with us.
+
+## Table of Contents
+
+- [Code of Conduct](#code-of-conduct)
+- [Getting Started](#getting-started)
+- [Development Setup](#development-setup)
+- [Contributing Guidelines](#contributing-guidelines)
+- [Pull Request Process](#pull-request-process)
+- [Testing](#testing)
+- [Documentation](#documentation)
+- [Community](#community)
+
+## Code of Conduct
+
+This project and everyone participating in it is governed by our [Code of Conduct](code-of-conduct.md). By participating, you are expected to uphold this code. Please report unacceptable behavior to [support@tensorlake.ai](mailto:support@tensorlake.ai).
+
+## Getting Started
+
+### Types of Contributions
+
+We welcome many types of contributions, including:
+
+- 🐛 **Bug fixes** - Help us squash bugs and improve reliability
+- ✨ **New features** - Add new functionality to the SDK
+- 📚 **Documentation** - Improve our docs, add examples, or write tutorials
+- 🧪 **Tests** - Add test coverage or improve existing tests
+- 🎨 **Examples** - Create new examples or improve existing ones
+- 🔧 **Developer experience** - Improve tooling, CI/CD, or development workflows
+
+### Before You Start
+
+1. **Search existing issues** - Check if someone else has already reported the bug or requested the feature
+2. **Create an issue** - For new features or significant changes, please create an issue first to discuss the approach
+3. **Start small** - If you're new to the project, consider starting with a "good first issue"
+
+## Development Setup
+
+### Prerequisites
+
+- Python 3.8 or higher
+- [Poetry](https://python-poetry.org/) for dependency management
+- Git
+
+### Setup Instructions
+
+1. **Fork and clone the repository**
+   ```bash
+   git clone https://github.com/YOUR_USERNAME/tensorlake.git
+   cd tensorlake
+   ```
+
+2. **Install dependencies**
+   ```bash
+   make build
+   ```
+
+3. **Set up environment variables**
+   ```bash
+   cp .env.example .env
+   # Edit .env with your API keys and configuration
+   ```
+
+4. **Activate the virtual environment**
+   ```bash
+   poetry shell
+   ```
+
+5. **Run tests to verify setup**
+   ```bash
+   make test
```

Comment:
`make test` won't work without running indexify. Let's only have doc ai make test here for now.